### PR TITLE
FOUR-20534: Implement the perfomance In-flight in Request and Cases

### PR DESCRIPTION
**Pull request by fagubla**
## Issue & Reproduction Steps
Implement the perfomance In-flight in Request and Cases

## Solution
- List the changes you've introduced to solve the issue.

## How to Test
Describe how to test that this solution works.

## Related Tickets & Packages
[FOUR-20534](https://processmaker.atlassian.net/browse/FOUR-20534)

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:deploy

[FOUR-20534]: https://processmaker.atlassian.net/browse/FOUR-20534?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by pmPaulis** [comment]
Team,
This PR has the issue reported here https://processmaker.atlassian.net/browse/SCO-26258
`An error occurred (AccessDeniedException) when calling the DescribeCluster operation: User: arn:aws:iam::***:user/*** is not authorized to perform: eks:DescribeCluster on resource: arn:aws:eks:***:***:cluster/pm4-eng`

Regards,

File: resources/jscomposition/cases/casesDetail/components/NewOverview.vue [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    });
+  }
+});
+
+onMounted(() => {
+  ProcessMaker.$modeler = modelerRef.value;
+});
+
+onBeforeUnmount(() => {
+  console.log("unmounted");
```
Comment (pmPaulis) [inline resources/jscomposition/cases/casesDetail/components/NewOverview.vue:132]: remove this console log

**Comment by pmPaulis** [comment]
@fagubla 
Review the overview from
- Cases 
- Request, this is showing a vue error:` [Vue warn]: Unknown custom element: <new-overview> - did you register the component correctly? For recursive components, make sure to provide the "name" option.`
Regards,

File: ProcessMaker/Http/Controllers/CasesController.php [hunk 2]
```diff
@@ -76,6 +78,45 @@ public function show($case_number)
         // Get the summary screen tranlations
         $this->summaryScreenTranslation($request);
 
+        // Load the process map
+        $processRequest = ProcessRequest::find($request->id);
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/CasesController.php:82]: You can improve this code used int RequestController Too like:
1. Create method loadProcessMap in ProcessMapTrait and use the instance here
```
        $inflightData = $this->loadProcessMap($request);
        $bpmn = $inflightData['bpmn'];
```
The method looks like:
```
/**
     * Load the process map and return inflight data.
     *
     * @param $request
     * @return array
     */
    private function loadProcessMap($request): array
    {
        $processRequest = ProcessRequest::find($request->id);
        $bpmn = $request->process->bpmn;
        $filteredCompletedNodes = [];
        $requestInProgressNodes = [];
        $requestIdleNodes = [];

        if ($processRequest) {
            $requestCompletedNodes = $processRequest->tokens()
                ->whereIn('status', ['CLOSED', 'COMPLETED', 'TRIGGERED'])
                ->pluck('element_id');
            $requestInProgressNodes = $processRequest->tokens()
                ->whereIn('status', ['ACTIVE', 'INCOMING'])
                ->pluck('element_id');

            // Remove any node that is 'ACTIVE' from the completed list.
            $filteredCompletedNodes = $requestCompletedNodes->diff($requestInProgressNodes)->values();

            // Obtain In-Progress nodes that were completed before
            $matchingNodes = $requestInProgressNodes->intersect($requestCompletedNodes);

            // Get idle nodes.
            $xml = $this->loadAndPrepareXML($bpmn);
            $nodeIds = $this->getNodeIds($xml);
            $requestIdleNodes = $nodeIds->diff($filteredCompletedNodes)->diff($requestInProgressNodes)->values();

            // Add completed sequence flow to the list of completed nodes.
            $sequenceFlowNodes = $this->getCompletedSequenceFlow($xml, $filteredCompletedNodes->implode(' '), $requestInProgressNodes->implode(' '), $matchingNodes->implode(' '));
            $filteredCompletedNodes = $filteredCompletedNodes->merge($sequenceFlowNodes);
        }

        return [
            'bpmn' => $bpmn,
            'requestCompletedNodes' => $filteredCompletedNodes,
            'requestInProgressNodes' => $requestInProgressNodes,
            'requestIdleNodes' => $requestIdleNodes,
            'requestId' => $request->process->id,
        ];
```

**Review by pmPaulis — CHANGES_REQUESTED** [review]

**Comment by fagubla** [comment]
> Team, This PR has the issue reported here https://processmaker.atlassian.net/browse/SCO-26258 `An error occurred (AccessDeniedException) when calling the DescribeCluster operation: User: arn:aws:iam::***:user/*** is not authorized to perform: eks:DescribeCluster on resource: arn:aws:eks:***:***:cluster/pm4-eng`
> 
> Regards,

@pmPaulis you need execute `npm run dev`

File: resources/jscomposition/cases/casesDetail/components/NewOverview.vue (hunk 1 above)
Comment (fagubla) [reply resources/jscomposition/cases/casesDetail/components/NewOverview.vue]: solved

File: ProcessMaker/Http/Controllers/CasesController.php (hunk 2 above)
Comment (fagubla) [reply ProcessMaker/Http/Controllers/CasesController.php]: solved

**Review by pmPaulis — APPROVED** [review]

**Pull request merged**